highfidelity/hifi
# Cleanup how Models do ray-picking and determination of point inside of convex hull

- Adds a `TriangleSet` class that does basic ray picking and convexHull point testing. This is essentially the same code as used to live directly in Model.cpp, but now it's been moved into a clean class where we can optimize the implementation without touching the Model class.
- Changes Model.cpp to use this new TriangleSet class.
- Changes Model.cpp to not constantly re-calculate the triangle list and mesh part boxes, and to instead simply use the details from the TriangleSets.

**NOTE:** because we were constantly recalculating the old triangle list every time the model moved or rotated, this new approach should see some slight performance improvement. This does not actually change the algorithm for doing ray-picking within a triangle set faster. That will be a subsequent PR.

------------
Test Plan:
- Smoke test
- Make sure to test Ray picking... e.g.:
    - using mouse to select things in edit.js
    - using hand controllers and lasers to click on things
- Make sure to test convex hull "zone shapes" to see if entering/leaving the zone still works correctly
